### PR TITLE
Welcome func_Wamley

### PR DESCRIPTION
**Pull request by LevanceWam**
Created a welcome functionality that will message the user when they first enter the server and when they hit the command !welcome.

### CODE REVIEW

**Comment by BHess2653** [comment]
LGTM!

**Comment by krogers78** [comment]
Looks good! I like the use of the username to personalize the message for each user!

File: bot/client.js [hunk 1]
```diff
@@ -57,4 +57,9 @@ client.on('message', (message) => {
   }
 });
 
+client.on('guildMemberAdd', (member) => {
+  // this will send a message to the user when they enter the discord sevrer
+  member.sendMessage(`Hey ${member.user.username} welcome to the server.`);
```
Comment (reactivepixel) [inline bot/client.js:62]: Consider that your implementation solves the problem, and elegantly, but falls just short in that your solution ​within the client.js file does not implement the controller you created. So in effect, any further maintenance will require duplicate efforts in keeping both messages identical.

File: bot/controllers/welcome.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    // Aliasing 'this' as controller to allow for binding in actions
+    const controller = this;
+
+    // Array of all commands, see baseCommand.js for prototype
+    this.commands = [
+      new Command(
+        '!welcome',
+        '!welcome',
+        'Welcome Message',
+        'Resend the welcome message you recieved when you joined the server.',
```
Comment (reactivepixel) [inline bot/controllers/welcome.js:19]: Spelling Error

File: bot/controllers/welcome.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        this.welcomeAction.bind(controller),
+        'dm',
+      ),
+    ];
+  }
+
+  // this message will be sent to the user's dm with their username
+  welcomeAction() {
+    const { message } = this;
+    return `Welcome to the server, ${message.author.username}`;
```
Comment (reactivepixel) [inline bot/controllers/welcome.js:29]: This message differs from the actual welcome message, so this is not actually a resending of that message. ITs these small deviation, combined with the help message that you are defining this as how you 'resend' that message that will confuse users.

**Review by reactivepixel — CHANGES_REQUESTED** [review]
Not ready to deploy as is. Spelling errors that are exposed to the user are not ideal.

Additionally, the controller issue will prevent this from being implemented in actuality.

**Pull request closed without merging**